### PR TITLE
GTC regions safety/a2b ord4

### PR DESCRIPTION
**Pull request by rheacangeo**
## Purpose

A2B_Ord4 has regions that depend on computations in neighboring regions. This can be a race condition and also is not efficient. Previous attempts to inline these same one resulted in memory issues, but that was with a more aggressive refactor of A2B_Ord4 to reduce the stencil count. Here we do the minimal necessary to unblock testing fv3core with regions in GTC.
It has been verified that this change does not slow does the code (https://jenkins.ginko.ch/job/fv3core-performance-test/573/backend=gtcuda,experiment=c128_6ranks_baroclinic,slave=daint_submit/artifact/2021-07-07-06-30-26.json)

## Code changes:

- a2b_ord4 has a light refactor to inline region calculations

## Infrastructure changes:

- Update the DEV=y setting with the Makefile to not docker build when that was set, as was intended with the previous dev_tests endpoint

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -156,7 +156,9 @@ dev_wrapper:
 
 test_base:
 ifneq ($(findstring docker,$(CONTAINER_CMD)),)
+    ifeq ($(DEV),n)
 	$(MAKE) build
+    endif
```
Comment (jdahm) [inline Makefile:161]: Is this change related?

File: fv3core/stencils/a2b_ord4.py [hunk 2]
```diff
@@ -285,15 +286,20 @@ def a2b_interpolation(
 
     with computation(PARALLEL), interval(...):
         qxx = a2 * (qx[0, -2, 0] + qx[0, 1, 0]) + a1 * (qx[0, -1, 0] + qx)
+        qyy = a2 * (qy[-2, 0, 0] + qy[1, 0, 0]) + a1 * (qy[-1, 0, 0] + qy)
+        # TODO use a function with an offset when that works consistently
```
Comment (jdahm) [inline fv3core/stencils/a2b_ord4.py:290]: Just a suggestion: We might want to start labeling todos with the person who wrote it, so that we can easily tell who to go to with questions about it. For example
```
# TODO(rheag): use a function with an offset when that works consistently
```

File: fv3core/stencils/a2b_ord4.py [hunk 3]
```diff
@@ -364,11 +389,20 @@ def qy_edge_south(qin: FloatField, dya: FloatFieldIJ):
 
 
 @gtscript.function
-def qy_edge_south2(qin: FloatField, dya: FloatFieldIJ, qy: FloatField):
+def qy_edge_south2(qin: FloatField, dya: FloatFieldIJ):
+    # TODO
```
Comment (jdahm) [inline fv3core/stencils/a2b_ord4.py:393]: Should this also be `# TODO: when possible`, or was there a reason not to add that here?

File: fv3core/stencils/a2b_ord4.py [hunk 4]
```diff
@@ -382,11 +416,22 @@ def qy_edge_north(qin: FloatField, dya: FloatFieldIJ):
 
 
 @gtscript.function
-def qy_edge_north2(qin: FloatField, dya: FloatFieldIJ, qy: FloatField):
+def qy_edge_north2(qin: FloatField, dya: FloatFieldIJ):
+    # TODO
```
Comment (jdahm) [inline fv3core/stencils/a2b_ord4.py:420]: Ditto.

**Review by jdahm — APPROVED** [review]
Looks good. Just the minor comments above.

File: Makefile (hunk 1 above)
Comment (rheacangeo) [reply Makefile:161]: it's not, I was hoping to get this into master because the DEV=y functionality needs this small tweak

File: fv3core/stencils/a2b_ord4.py (hunk 3 above)
Comment (rheacangeo) [reply fv3core/stencils/a2b_ord4.py:393]: I think it's implied by the other TODO in the same pattern, but I can change it.

**Comment by rheacangeo** [comment]
Wow, here's a fun new error
MPICH2 ERROR [Rank 4] [job id 32387295] [Wed Jul  7 08:18:02 2021] [c0-3c2s0n0] [nid05888] - MPID_nem_gni_check_localCQ(): GNI_CQ_EVENT_TYPE_POST had error (SOURCE_SSID:AT_MDD_INV:CPLTN_DREQ)

File: fv3core/stencils/a2b_ord4.py (hunk 2 above)
Comment (rheacangeo) [reply fv3core/stencils/a2b_ord4.py:290]: We can, though it's possible to tell who wrote each line of code in the git history

**Comment by rheacangeo** [comment]
launch jenkins
